ios/BT/ExposureManager.swift

```diff
@@ -346,18 +341,38 @@ final class ExposureManager: NSObject {
                               completionHandler: completionHandler)
                   return
                 }
-                let newExposures = (exposures ?? []).map { exposure in
-                  Exposure(id: UUID().uuidString,
-                           date: exposure.date.posixRepresentation,
-                           duration: exposure.duration,
-                           totalRiskScore: exposure.totalRiskScore,
-                           transmissionRiskLevel: exposure.transmissionRiskLevel)
+                if let summary = summary, ExposureManager.score(summary: summary, with: configuration) {
```

Comment:
Noticing this method is getting pretty gnarly, idk what is idiomatic swfit, but should be be pulling some of this out into named helpers/functions?

Reply:
yes John, im thinking about adding a promise library to cope with this method complexity, but I feel first we have to get it working correctly with a fair amount of Unit Tests, what do you think?